internal/pkg/cli/app_show.go

```diff
@@ -72,109 +62,100 @@ type ShowAppOpts struct {
 }
 
 // Ask asks for fields that are required but not passed in.
-func (opts *ShowAppOpts) Ask() error {
-	// get project
-	if opts.ProjectName() == "" {
-		projectName, err := opts.selectProject()
+func (o *ShowAppOpts) Ask() error {
+	if err := o.askProject(); err != nil {
+		return err
+	}
+	return o.askAppName()
+}
+
+// Validate returns an error if the values provided by the user are invalid.
+func (o *ShowAppOpts) Validate() error {
+	if o.ProjectName() != "" {
+		names, err := o.retrieveProjects()
 		if err != nil {
 			return err
 		}
-		opts.projectName = projectName
+		if !contains(o.ProjectName(), names) {
+			return fmt.Errorf("project '%s' does not exist in the workspace", o.ProjectName())
+		}
```

Comment:
Are we prompting for a project name and then returning an error if the selected project name isn't part of the local workspace? Should we instead simply error out if a local workspace isn't detected?

Reply:
I think this is if someone passes in a project flag.

Reply:
So I had the same question and had an offline discussion with Efe. We think for `show` it might not be necessary for users to stay in a local workspace. If they are not in a local ws then we can prompt them to select the project. What do you think?

Reply:
But this is forcing you to be in a workspace, right? `o.ProjectName()` is populated by being in a local workspace?

Reply:
Not necessary I think, if you are not in a workspace it will prompt you to choose a project.

Reply:
Oh yeah I think you're right. I keep thinking `Validate` happens after `Ask`, but it's the other way around.